src/mam/mod_mam_muc_rdbms_arch.erl

```diff
@@ -112,6 +115,11 @@ register_prepared_queries() ->
     mongoose_rdbms:prepare(mam_muc_archive_remove, mam_muc_message, [room_id],
                            <<"DELETE FROM mam_muc_message "
                              "WHERE room_id = ?">>),
+    mongoose_rdbms:prepare(mam_muc_remove_domain, mam_muc_message, ['mam_server_user.server'],
+                           <<"DELETE FROM mam_muc_message "
+                             "WHERE room_id IN (SELECT id FROM mam_server_user where server = ?)">>),
```

Comment:
how optimal is this query? what if we have over 100k records selected by `SELECT id FROM mam_server_user where server = ?`

Reply:
it should work.

ids would go into a temp table in this case. And use similar logic as left joins.

 It would be a pretty long transaction though.